addons/binding/org.openhab.binding.xmltv/src/main/java/org/openhab/binding/xmltv/internal/handler/ChannelHandler.java

```diff
@@ -0,0 +1,226 @@
+            globalJob = scheduler.scheduleWithFixedDelay(() -> {
+                if (programmes.size() < 2) {
+                    refreshProgramList();
+                }
+                if (programmes.size() == 0) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE,
+                            "No programs to come in the current XML file for this channel");
+                } else if (Instant.now().isAfter(programmes.get(0).getProgrammeStop())) {
+                    programmes.remove(0);
+                    programIcon = downloadIcon(programmes.get(0).getIcons());
+                }
+
+                getThing().getChannels().forEach(channel -> updateChannel(channel.getUID()));
+
+            }, 3, config.refresh, TimeUnit.SECONDS);
+        }
+    }
+
+    private void refreshProgramList() {
+        Bridge bridge = getBridge();
+        if (bridge != null && bridge.getStatus() == ThingStatus.ONLINE) {
+            Tv tv = ((XmlTVHandler) bridge.getHandler()).getXmlFile();
+            if (tv != null) {
+                String channelId = (String) getConfig().get(XmlChannelConfiguration.CHANNEL_ID);
+
+                if (mediaChannel == null) {
+                    Optional<MediaChannel> channel = tv.getMediaChannels().stream()
+                            .filter(mediaChannel -> mediaChannel.getId().equals(channelId)).findFirst();
+                    if (channel.isPresent()) {
+                        mediaChannel = channel.get();
+                        mediaIcon = downloadIcon(mediaChannel.getIcons());
+                    }
+                }
+
+                programmes.clear();
+                tv.getProgrammes().stream()
+                        .filter(p -> p.getChannel().equals(channelId) && p.getProgrammeStop().isAfter(Instant.now()))
+                        .forEach(p -> programmes.add(p));
+
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "No file available");
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (globalJob != null && !globalJob.isCancelled()) {
+            globalJob.cancel(true);
+            globalJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // TODO Auto-generated method stub
+    }
+
+    /**
+     * Update the channel from the last OpenUV data retrieved
+     *
+     * @param channelUID the id identifying the channel to be updated
+     *
+     */
+    private void updateChannel(ChannelUID channelUID) {
+        String[] uidElements = channelUID.getId().split("#");
+        if (uidElements.length == 2) {
+            int target = GROUP_NEXT_PROGRAM.equals(uidElements[0]) ? 1 : 0;
+            Programme programme = programmes.get(target);
```

Comment:
According to this [community post](https://community.openhab.org/t/use-xmltv-with-openhab/60367/5?u=wborn) `programmes` can be empty whereas the `target` can be 1 resulting in an `IndexOutOfBoundsException`:

```
2018-12-21 00:31:12.335 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.util.ArrayList.rangeCheck(ArrayList.java:657) ~[?:?]
	at java.util.ArrayList.get(ArrayList.java:433) ~[?:?]
	at org.openhab.binding.xmltv.internal.handler.ChannelHandler.updateChannel(ChannelHandler.java:151) ~[?:?]
	at org.openhab.binding.xmltv.internal.handler.ChannelHandler.lambda$1(ChannelHandler.java:92) ~[?:?]
	at java.util.ArrayList.forEach(ArrayList.java:1257) ~[?:?]
	at java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1080) ~[?:?]
	at org.openhab.binding.xmltv.internal.handler.ChannelHandler.lambda$0(ChannelHandler.java:92) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:?]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) ~[?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) ~[?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
```